core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -692,4 +697,15 @@ private RequestId nextRequestId() {
             return id;
         }
     }
+
+    void swapServerConfig(ServerConfig config) {
+        requireNonNull(config);
+        if (this.config != config) {
+            this.config = config;
+        }
```

Comment:
No need to check identity when assigning:
```suggestion
        this.config = config;
```